conan/cli/commands/create.py

```diff
@@ -50,8 +52,12 @@ def create(conan_api, parser, *args):
                                   user=args.user, channel=args.channel,
                                   lockfile=lockfile)
     if lockfile:
-        # FIXME: We need to update build_requires too, not only ``requires``
-        lockfile.add(requires=[ref])
+        if args.python_require:
+            lockfile.add(python_requires=[ref])
+        elif args.build_require:
+            lockfile.add(build_requires=[ref])
+        else:
+            lockfile.add(requires=[ref])
```

Comment:
Should we add a test to check this?

Reply:
Yes, I'll do it

Reply:
Removed this thing from this PR, will do another PR with more stuff related to lockfiles there are things missing there